src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/OverloadResolution.cs

```diff
@@ -323,9 +323,9 @@ private static bool OverloadResolutionResultIsValid<TMember>(ArrayBuilder<Member
             ref HashSet<DiagnosticInfo> useSiteDiagnostics)
         {
             Debug.Assert(funcPtrBuilder.Count == 1);
+            Debug.Assert(funcPtrBuilder[0].Arity == 0);
```

Comment:
The only time `inferWithDynamic` is used is when `Arity` is `> 0`, so I've added an assert that that's never true (which it can't be, as function pointers cannot define their own type parameters).